src/System.Drawing.Common/src/System/Drawing/Printing/PrintingServices.Unix.cs

```diff
@@ -589,7 +609,15 @@ private static PrinterResolution ParseResolution(string resolution)
         /// </summary>
         private static void LoadPrinters()
         {
-            installed_printers.Clear();
+            lock (installed_printers)
+            {
+                installed_printers.Clear();
+            }
+            lock (populatedLock)
```

Comment:
I don't think the boolean should have its own lock - also you have used it below (line 672) without a lock. I think the boolean should always be read and written within the `installed_printers` lock.